### PR TITLE
workflows: build: Add Meson build job

### PR DESCRIPTION
**Pull request by ernestask**
This commit adds a parallel job to run Meson test builds.

### CODE REVIEW

**Comment by ernestask** [comment]
Seems like a misplaced `-Werror` is causing the Python module to not build (SWIG spits out some not-nice code).

**Comment by dgibson** [comment]
@ernestask in the github action itself I'm seeing setuptools errors before we even get to swig.

**Comment by ernestask** [comment]
Oh, right, that is caused by not having a recent-enough git. If you check the logs for the checkout step, you’ll see a warning.

**Comment by ernestask** [comment]
But this is a draft anyway, I’ll continue tinkering today.

**Comment by ernestask** [comment]
A couple of hacks later, it seems to be building.

- setuptools_scm requires a git repo, but not having git installed when using `actions/checkout` results in a source archive being fetched and exploded. I fixed that by adding a step before checkout to install git.

- Later, Ubuntu builds were failing as described in https://github.com/actions/checkout/issues/766, so another step was required to add $GITHUB_WORKSPACE to safe.directory in .gitconfig.
 
Make builds are not affected, because they use a different kludge (`SETUPTOOLS_SCM_PRETEND_VERSION="0"`).

**Comment by dgibson** [comment]
> Oh, right, that is caused by not having a recent-enough git. If you check the logs for the checkout step, you’ll see a warning.

Uh.. not recent enough which git?  I'm literally working from the dtc main branch...

**Comment by ernestask** [comment]
> Uh.. not recent enough which git? I'm literally working from the dtc main branch...

I mean in the Actions runner. Currently you use a base container image for multiple distros, none of which have git installed.

**Comment by dgibson** [comment]
> > Uh.. not recent enough which git? I'm literally working from the dtc main branch...
> 
> I mean in the Actions runner. Currently you use a base container image for multiple distros, none of which have git installed. My phrasing could have been better.

Ah!  Now I see.

**Comment by SoldierC4** [comment]
> A couple of hacks later, it seems to be building.
> 
> * setuptools_scm requires a git repo, but not having git installed when using `actions/checkout` results in a source archive being fetched and exploded. I fixed that by adding a step before checkout to install git.
> * Later, Ubuntu builds were failing as described in [fatal: unsafe repository (REPO is owned by someone else) in other workflow steps after running checkout actions/checkout#766](https://github.com/actions/checkout/issues/766), so another step was required to add $GITHUB_WORKSPACE to safe.directory in .gitconfig.
> 
> Make builds are not affected, because they use a different kludge (`SETUPTOOLS_SCM_PRETEND_VERSION="0"`).

I opened a pull request to fix the setuptools_scm issue so a git repo is no longer required https://github.com/dgibson/dtc/pull/114

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -35,3 +36,45 @@ jobs:
...
+      - name: Install dependencies
+        run: "${GITHUB_WORKSPACE}/scripts/install-deps.sh"
+
+      # https://github.com/actions/checkout/issues/766
+      - name: Add workspace to safe.directory
+        run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+      - name: Build
+        run: |
+          meson setup . _build
```
Comment (SoldierC4) [inline .github/workflows/build.yml:79]: Once Meson deprecations are fixed in https://github.com/dgibson/dtc/pull/113, we can use `meson setup --fatal-meson-warnings` to enforce version warnings.

**Comment by dgibson** [comment]
@ernestask as @blmaier notes we have some fixes on the way which should make this worth rebasing.

Note that before I can apply, I'll also need at least Signed-off-by lines on each of the patches in this series.  Most of the patches will also need a more substantial commit message giving the rationale for each change (just a subject line is almost never sufficient).

**Pull request closed without merging**